ClassHierarchy_hw/Program.cs

```diff
@@ -0,0 +1,204 @@
+
+            // Calling the interface
+            SaladBuilderInterface(ingredientTypes, ingredients);
+        }
+
+        // A method that creates categories
+        public static void CreateIngredientTypes(ref IngredientType[] ingredientTypes, IngredientType ingredientType)
+        {
+            ArrayPush(ref ingredientTypes, ingredientType);
+        }
+
+        // A method that creates ingredients
+        public static void CreateIngredients(ref Ingredient[] ingredientTypes, Ingredient ingredientType)
+        {
+            ArrayPush(ref ingredientTypes, ingredientType);
+        }
+
+        // Interface method to ask a person about what ingredients to add and returns a salad
+        public static void SaladBuilderInterface(IngredientType[] ingredientTypes, Ingredient[] ingredients)
+        {
+            int optionCounter = 0;
+            Console.WriteLine(" Hi this is a SALAD BUILDER NIKITA 3000");
+            Console.WriteLine("///////////");
+            Console.WriteLine(" Let's build our own salad!");
+            Console.WriteLine("///////////");
+            Console.WriteLine(" Please choose your own content of the salad");
+            Console.WriteLine(" You can choose from: ");
+            Console.WriteLine("//////////");
+            foreach (IngredientType ingredientType in ingredientTypes)
+            {
+                Console.WriteLine($"{ingredientType.Name} - Max {ingredientType.MaxQuantity}");
+                for (int i = 0; i < ingredients.Length; i++)
+                {
+                    Ingredient ingred = ingredients[i];
+                    if (ingred.Type == ingredientType.Name)
+                    {
+                        optionCounter++;
+                        Console.WriteLine($" {optionCounter} - {ingred.Name}");
+                    }
+                }
+            }
+
+            Console.WriteLine("Please enter your choice.");
+            string ingredientNumInput = Console.ReadLine();
+            int n;
+            bool isNumeric = int.TryParse(ingredientNumInput, out n);
+
+            if (isNumeric && n <= optionCounter && n != 0 && ingredientNumInput != null)
+            {
+                Ingredient ing = ingredients[n - 1];
+                Console.WriteLine("Provide us with some measurements");
+                string measurement = Console.ReadLine();
+                if (measurement != "oz" && measurement != "gramm" && measurement != "cup")
+                {
+                    WrongInput(ingredientTypes, ingredients);
+                }
+                else
+                {
+                    ing.Measurement = measurement;
+                }
+
+                CreateSalad(ingredientTypes, ingredients, ing);
+            }
+            else
+            {
+                WrongInput(ingredientTypes, ingredients);
+            }
+        }
+
+        public static void WrongInput(IngredientType[] ingredientTypes, Ingredient[] ingredients)
```

Comment:
Main method look way too heavy, consider creating separate class for the functionality with console + a separate one for Salad management flow